File: packages/core/tests/fixtures/python/sample.py
"""
Sample Python file for testing
"""

import os
import sys
from typing import List, Dict, Optional

class DataProcessor:
    """A sample data processor class"""
    
    def __init__(self, config: Dict[str, str]):
        self.config = config
        self._cache = {}
    
    def process_data(self, data: List[str]) -> List[str]:
        """Process a list of data items"""
        results = []
        
        for item in data:
            if item and len(item) > 0:
                processed = self._process_item(item)
                results.append(processed)
        
        return results
    
    def _process_item(self, item: str) -> str:
        """Private method to process individual items"""
        return item.upper()
    
    def get_cache_size(self) -> int:
        """Get the current cache size"""
        return len(self._cache)

def calculate_fibonacci(n: int) -> int:
    """Calculate the nth Fibonacci number"""
    if n <= 1:
        return n
    
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    
    return b

async def async_operation(data: str) -> str:
    """An async function for testing"""
    await asyncio.sleep(0.1)
    return data.upper()

# Private function
def _helper_function():
    """This is a private helper function"""
    pass

# Module-level variable
API_VERSION = "1.0.0"

File: packages/core/tests/fixtures/go/sample.go
package main

import (
	"fmt"
	"strings"
	"time"
)

// DataProcessor handles data processing operations
type DataProcessor struct {
	config map[string]string
	cache  map[string]interface{}
}

// NewDataProcessor creates a new DataProcessor instance
func NewDataProcessor(config map[string]string) *DataProcessor {
	return &DataProcessor{
		config: config,
		cache:  make(map[string]interface{}),
	}
}

// ProcessData processes a slice of strings
func (dp *DataProcessor) ProcessData(data []string) []string {
	results := make([]string, 0, len(data))
	
	for _, item := range data {
		if len(item) > 0 {
			processed := dp.processItem(item)
			results = append(results, processed)
		}
	}
	
	return results
}

// processItem is a private method
func (dp *DataProcessor) processItem(item string) string {
	return strings.ToUpper(item)
}

// GetCacheSize returns the current cache size
func (dp *DataProcessor) GetCacheSize() int {
	return len(dp.cache)
}

// CalculateFibonacci calculates the nth Fibonacci number
func CalculateFibonacci(n int) int {
	if n <= 1 {
		return n
	}
	
	a, b := 0, 1
	for i := 2; i <= n; i++ {
		a, b = b, a+b
	}
	
	return b
}

// ProcessComplexData handles complex data processing
func ProcessComplexData(input []string) ([]string, error) {
	if len(input) == 0 {
		return nil, fmt.Errorf("input cannot be empty")
	}
	
	results := make([]string, 0, len(input))
	
	for i, item := range input {
		switch {
		case strings.HasPrefix(item, "A"):
			results = append(results, processTypeA(item))
		case strings.HasPrefix(item, "B"):
			results = append(results, processTypeB(item))
		default:
			results = append(results, fmt.Sprintf("unknown_%d", i))
		}
	}
	
	return results, nil
}

// processTypeA handles type A items
func processTypeA(item string) string {
	if len(item) > 5 {
		return fmt.Sprintf("A_LONG_%s", item)
	}
	return fmt.Sprintf("A_SHORT_%s", item)
}

// processTypeB handles type B items
func processTypeB(item string) string {
	return fmt.Sprintf("B_%s", strings.ToUpper(item))
}

// privateHelper is a private function
func privateHelper() {
	fmt.Println("This is a private helper function")
}

// Constants
const (
	API_VERSION = "1.0.0"
	MAX_RETRIES = 3
)
